capstone/src/main/webapp/scripts/search.js

```diff
@@ -0,0 +1,76 @@
+const searchInput = document.getElementById("name-search");
+const suggestionsPanel = document.getElementById("suggestions");
+
+let names = [
+  { fullname: "Lucy Qu" },
+  { fullname: "Lucy Liu" },
+  { fullname: "Himani Yadav" },
+  { fullname: "Anika Bagga" },
+  { fullname: "Lucille Ball" }
+];
+let currentFocus = -1;
+
+function autocomplete() {
+  searchInput.addEventListener("keyup", filter);
+  searchInput.addEventListener("keyup", checkKey);
+}
+
+function filter() {
+  const input = searchInput.value.toLowerCase();
+  suggestionsPanel.innerHTML = "";
+  const suggestions = names.filter(function(name) {
+    return name.fullname.toLowerCase().startsWith(input);
+  });
+  suggestions.forEach(suggested => appendSuggestion(suggested));
+  if (input === "") {
+    suggestionsPanel.innerHTML = "";
+  }
+}
+
+function checkKey(e) {
+  let list = document.getElementById("suggestions");
+  if (list) {
+    list = list.getElementsByTagName("div");
+  }
+  if (e.keyCode == "38") {
```

Comment:
I know you have comments but can you add functions for the different key codes that describe what that key press is? i.e. isUpArrow, isDownArrow, isEnter, etc.? It'd be easier to read that way and not rely on the comment since it is awkwardly below the conditional

Reply:
^ That or define constants at the top of the file:

```javascript
const UP_ARROW_KEYCODE = "38";
```

or

```javascript
const KeyCodes = {
  UP: "38",
  DOWN: ...
};
```